lib/embulk/output/bigquery.rb

```diff
@@ -231,10 +232,42 @@ def self.configure(config, schema, task_count)
           unless task['time_partitioning']['type']
             raise ConfigError.new "`time_partitioning` must have `type` key"
           end
-        elsif Helper.has_partition_decorator?(task['table'])
+        # If user specify range_partitioning, it should be used as is
+        elsif Helper.has_partition_decorator?(task['table']) && task['range_partitioning'].nil?
           task['time_partitioning'] = {'type' => 'DAY'}
         end
 
+        if task['range_partitioning']
+          unless task['range_partitioning']['field']
+            raise ConfigError.new "`range_partitioning` must have `field` key"
+          end
+          unless task['range_partitioning']['range']
+            raise ConfigError.new "`range_partitioning` must have `range` key"
+          end
+
+          unless task['range_partitioning']['range']['start']
+            raise ConfigError.new "`range_partitioning` must have `range.start` key"
+          end
+          unless task['range_partitioning']['range']['start'].is_a?(Integer)
+            raise ConfigError.new "`range_partitioning.range.start` must be an integer"
+          end
+          unless task['range_partitioning']['range']['end']
+            raise ConfigError.new "`range_partitioning` must have `range.end` key"
+          end
+          unless task['range_partitioning']['range']['end'].is_a?(Integer)
+            raise ConfigError.new "`range_partitioning.range.end` must be an integer"
+          end
+          if task['range_partitioning']['range']['start'] >= task['range_partitioning']['range']['end']
```

Comment:
As I said before, checking the range as `start` + `interval` < `end` would be best. What do you think? (This is my opinion)